roboteam_ai/src/coach/heuristics/OffensiveScore.cpp

```diff
@@ -11,24 +12,49 @@ namespace coach {
 OffensiveScore g_offensiveScore;
 
 /// Calculates a total score based on all the sub-scores
-double OffensiveScore::calculateOffensivePositionScore(const Vector2 &position) {
-    WorldData world = world::world->getWorld();
-    roboteam_msgs::GeometryFieldSize field = world::field->get_field();
+double OffensiveScore::calculateOffensivePositionScore(const Vector2 &zoneLocation, const Vector2 &position,
+        const WorldData &world, const roboteam_msgs::GeometryFieldSize &field) {
+
+    if (!positionIsValid(zoneLocation, position)) return 0.0;
 
     double closeToGoalScore = CoachHeuristics::calculateCloseToGoalScore(position);
     double passLineScore = CoachHeuristics::calculatePassLineScore(position, world);
     double shotAtGoalScore = CoachHeuristics::calculateShotAtGoalScore(position, world);
     double distanceToOpponentScore = CoachHeuristics::calculateDistanceToOpponentsScore(position, world);
     double distanceToBallScore = CoachHeuristics::calculateDistanceToBallScore(position, world);
 
-    double score =  SHOT_AT_GOAL_WEIGHT * shotAtGoalScore +
-                    PASS_LINE_WEIGHT * passLineScore +
-                    CLOSE_TO_GOAL_WEIGHT * closeToGoalScore +
-                    DISTANCE_TO_OPPONENT_WEIGHT * distanceToOpponentScore +
-                    DISTANCE_TO_BALL_WEIGHT * distanceToBallScore;
+    double score = SHOT_AT_GOAL_WEIGHT*shotAtGoalScore +
+            PASS_LINE_WEIGHT*passLineScore +
+            CLOSE_TO_GOAL_WEIGHT*closeToGoalScore +
+            DISTANCE_TO_OPPONENT_WEIGHT*distanceToOpponentScore +
+            DISTANCE_TO_BALL_WEIGHT*distanceToBallScore;
     return score;
 }
 
+bool OffensiveScore::positionIsValid(const Vector2 &zoneLocation, const Vector2 &positionToCheck) {
```

Comment:
I think its weird to have a positionIsValid function when you pick positions itself. What exactly does it check?